reactor-core/src/main/java/reactor/core/publisher/FluxDistinctUntilChanged.java

```diff
@@ -103,8 +111,7 @@ public boolean tryOnNext(T t) {
 				return true;
 			}
 
-			if (Objects.equals(lastKey, k)) {
-				lastKey = k;
+			if (null != lastKey && keyComparator.test(lastKey, k)) {
 				return false;
```

Comment:
When `lastKey` is null (first time around), I do not invoke the comparator as we know that the value is distinct. I broke apart the conditional to make that more explicit when adding the error handling